merlion/models/utils/rolling_window_dataset.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from typing import Optional
+from merlion.utils.time_series import TimeSeries
+
+
+class RollingWindowDataset:
+
+    def __init__(
+            self,
+            data: TimeSeries,
+            target_seq_index: int,
+            maxlags: int,
+            forecast_steps: int,
+            shuffle: bool = False,
+            ts_index: bool = False,
```

Comment:
I think you should remove the `ts_index` option altogether. I included it in the example code I sent you to show that it has poor performance.

Reply:
`ts_index = False` is used in most case. I have the `ts_index = True` for the scenario that user wants to accept TimeSeries and output TimeSeries, in this case, I have this option so TimeSeries will keep as its type, otherwise it might be converted multiple types between TimeSeries and pandas anyway. One use case is in my unittest, I will use TimeSeries to feed train and forecast APIs explictly, in this way, I do not need to convert type.